enterprise/app/api_keys/api_keys.tsx

```diff
@@ -315,13 +321,13 @@ export default class ApiKeysComponent extends React.Component<ApiKeysComponentPr
     title: string;
     submitLabel: string;
     onRequestClose: () => any;
-    onSubmit: () => any;
+    onSubmit: (e: React.FormEvent) => any;
     onChange: (name: string, value: any) => any;
     ref: React.RefObject<HTMLFormElement>;
     formState: FormState<T>;
   }) {
     return (
-      <Modal isOpen={isOpen} onRequestClose={onRequestClose} shouldFocusAfterRender={false}>
+      <Modal isOpen={Boolean(isOpen)} onRequestClose={onRequestClose} shouldFocusAfterRender={false}>
```

Comment:
I guess i'm surprised you need the cast here? Doesn't the destructuring carry the type info along with it?

Reply:
Good catch, I made this change before making all the fields required.